google/cloud/opentelemetry/internal/recordable.cc

```diff
@@ -184,9 +204,19 @@ void Recordable::SetResource(
     opentelemetry::sdk::resource::Resource const& /*resource*/) noexcept {}
 
 void Recordable::SetStartTime(
-    opentelemetry::common::SystemTimestamp /*start_time*/) noexcept {}
+    opentelemetry::common::SystemTimestamp start_time) noexcept {
+  // std::chrono::system_clock may not have nanosecond resolution on some
+  // platforms, so we avoid using it for conversions between OpenTelemetry
+  // time and Protobuf time.
```

Comment:
Looks good.  It seems like `opentelemetry::common::SystemTimestamp`'s implicit conversion to `std::chrono::system_clock::time_point` is a rounding accident waiting to happen.

Reply:
Yeah, thanks for the warning.